src/baseWidgetView.ts

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Trung Le
+// Distributed under the terms of the Modified BSD License.
+
+import { DOMWidgetView, WidgetView } from '@jupyter-widgets/base';
+import { IThemeManager } from '@jupyterlab/apputils';
+import { Debouncer } from '@lumino/polling';
+import * as echarts from 'echarts';
+import 'echarts-gl';
+import { isLightTheme } from './tools';
+
+export abstract class BaseEChartsWidgetView extends DOMWidgetView {
+  initialize(parameters: WidgetView.IInitializeParameters): void {
+    super.initialize(parameters);
+    this.setupThemeListener();
+    this.setupResizeListener();
+
+    // Define arrays of properties
+    const initProps = [
+      'theme',
+      'device_pixel_ratio',
+      'renderer',
+      'use_dirty_rect',
+      'use_coarse_pointer',
+      'pointer_size',
+      'width',
+      'height',
+      'locale'
+    ];
+    // Set up listeners for init properties
+    initProps.forEach(prop => {
+      this.model.on(`change:${prop}`, this.recreateChart, this);
+    });
+    this.model.on('change:style', this.setStyle, this);
+    this.model.on('change', this.value_changed, this);
+  }
```

Comment:
`this.value_changed` will also be called on the init props change. This might make to redundant chart re-rendering. You  need to do some event filtering in the `value_changed` callback. 
At the same time I can not recall why i did not follow the camelCase naming here. Could you update it?

Reply:
I'm struggling with this one, `this.model` doesn't show the props that have changed from the inner models(?), i.e., if `Line.data` changes, `this.model` doesn't show the change

Reply:
I pushed the changes needed to filter the events. Please check if it works as intended.

Reply:
thanks!!